tests/image_test.py

```diff
@@ -0,0 +1,176 @@
+                                  antialias):
+    # TODO(phawkins): debug this. There is a small mismatch between TF and JAX
+    # for some cases of non-antialiased bicubic downscaling; we would expect
+    # exact equality.
+    if method == "bicubic" and any(x < y for x, y in
+                                   zip(target_shape, image_shape)):
+      raise unittest.SkipTest("non-antialiased bicubic downscaling mismatch")
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: (rng(image_shape, dtype),)
+    def tf_fn(x):
+      out = tf.image.resize(x, tf.constant(target_shape[1:-1]), method=method,
+                             antialias=antialias).numpy()
+      return out
+    jax_fn = partial(image.resize, shape=target_shape, method=method,
+                     antialias=antialias)
+    self._CheckAgainstNumpy(tf_fn, jax_fn, args_maker, check_dtypes=True,
+                            tol=1e-4)
+
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+       {"testcase_name": "_shape={}_target={}_method={}".format(
+          jtu.format_shape_dtype_string(image_shape, dtype),
+          jtu.format_shape_dtype_string(target_shape, dtype), method),
+        "dtype": dtype, "image_shape": image_shape,
+        "target_shape": target_shape,
+        "method": method}
+       for dtype in [np.float32]
+       for target_shape, image_shape in itertools.combinations_with_replacement(
+        [[3, 2], [6, 4], [33, 17], [50, 38]], 2)
+       for method in ["bilinear", "lanczos3", "bicubic"]))
+  @unittest.skipIf(not PIL_Image, "Test requires PIL")
+  def testResizeAgainstPIL(self, dtype, image_shape, target_shape, method):
+    rng = jtu.rand_uniform(self.rng())
+    args_maker = lambda: (rng(image_shape, dtype),)
+    def pil_fn(x):
+      pil_methods = {
+        "bilinear": PIL_Image.BILINEAR,
+        "bicubic": PIL_Image.BICUBIC,
+        "lanczos3": PIL_Image.LANCZOS,
+      }
+      img = PIL_Image.fromarray(x)
+      out = np.asarray(img.resize(target_shape[::-1], pil_methods[method]))
+      return out
+    jax_fn = partial(image.resize, shape=target_shape, method=method,
+                     antialias=True)
+    self._CheckAgainstNumpy(pil_fn, jax_fn, args_maker, check_dtypes=True,
+                            tol=1e-4)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+       {"testcase_name": "_shape={}_target={}_method={}".format(
+          jtu.format_shape_dtype_string(image_shape, dtype),
+          jtu.format_shape_dtype_string(target_shape, dtype), method),
+        "dtype": dtype, "image_shape": image_shape, "target_shape": target_shape,
+        "method": method}
+       for dtype in inexact_dtypes
+       for image_shape, target_shape in [
+         ([3, 1, 2], [6, 1, 4]),
+         ([1, 3, 2, 1], [1, 6, 4, 1]),
+       ]
+       for method in ["linear", "lanczos3", "lanczos5", "cubic"]))
+  def testResizeUp(self, dtype, image_shape, target_shape, method):
```

Comment:
Is there a testResizeDown, or is that implicit in the other tests.

Reply:
No, not at the moment. As you may have noticed, I ported `testResizeUp` from TensorFlow (ok in this case since it is Google-authored).

However I found that the corresponding `testResizeDown` actually seems not to apply to any of these resizes because the TF test isn't actually testing any methods except for nearest-neighbor. The logic here:
https://github.com/tensorflow/tensorflow/blob/9065899e9252d5d9472b45d5a3dbecfb8b039117/tensorflow/python/ops/image_ops_test.py#L2511
only enables that test for resizes that run on GPU, but only nearest neighbor is implemented on GPU.

Since I didn't have a great deal of confidence in producing correct golden outputs, I thought it safer just to test against PIL and TensorFlow. Note that those tests *do* test downsampling because they make a random selection of input and output shapes.